source/backend/dal/Repositories/PropertyRepository.cs

```diff
@@ -464,6 +465,74 @@ public HashSet<long> GetMatchingIds(PropertyFilterCriteria filter)
             return query.Select(p => p.PropertyId).ToHashSet();
         }
 
+        /// <summary>
+        /// Return a summary List of Management activities for a especific property.
+        /// </summary>
+        /// <param name="propertyId"></param>
+        /// <returns>List of Property's management activities.</returns>
+        public IList<PimsPropPropActivity> GetManagementActivitiesByProperty(long propertyId)
+        {
+            User.ThrowIfNotAllAuthorized(Permissions.ManagementView);
+
+            List<PimsPropPropActivity> activities = Context.PimsPropPropActivities.AsNoTracking()
+                    .Include(x => x.PimsPropertyActivity)
+                        .ThenInclude(y => y.PropMgmtActivityTypeCodeNavigation)
+                    .Include(x => x.PimsPropertyActivity)
+                        .ThenInclude(y => y.PropMgmtActivitySubtypeCodeNavigation)
+                    .Include(x => x.PimsPropertyActivity)
+                       .ThenInclude(y => y.PropMgmtActivityStatusTypeCodeNavigation)
+                    .Where(p => p.PropertyId == propertyId)
+                    .ToList();
+
+            return activities;
+        }
+
+        /// <summary>
+        /// Get the Property Management Activity by Id.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public PimsPropPropActivity GetManagementActivityById(long id)
+        {
+            User.ThrowIfNotAllAuthorized(Permissions.ManagementView);
+
+            return Context.PimsPropPropActivities
+                        .AsNoTracking()
+                        .Include(x => x.PimsPropertyActivity)
+                        .FirstOrDefault(p => p.PropPropActivityId == id) ?? throw new KeyNotFoundException();
+        }
+
+        /// <summary>
+        /// TryDelete the Activity asscociated with the property and if no property associated to activity delete the activicy as well.
+        /// </summary>
+        /// <param name="managementActivityId"></param>
+        /// <returns>Boolean of deletion sucess.</returns>
+        public bool TryDeletePropertyActivity(long managementActivityId)
+        {
+            bool deletedSuccessfully = false;
+            var deletedEntity = Context.PimsPropPropActivities.FirstOrDefault(x => x.PropPropActivityId == managementActivityId);
+
+            if (deletedEntity is not null)
+            {
+                if (Context.PimsPropPropActivities.Count(x => x.PimsPropertyActivityId == deletedEntity.PimsPropertyActivityId) > 1)
```

Comment:
can you explain what you are trying to do here? it seems like you are manually implement a cascade here but I believe we already have a db cascade in place for PimsPropertyActivity and PimsPropPropActivity. This top if statement also appears to be only removing the many-to-many, that seems odd if the caller requested that an activity by deleted by an activity id, as the activity will still be in the system, just not the first many-to-many with a matching id.

Reply:
As discussed, updated method name and added comments for clarity.